mkosi

```diff
@@ -469,6 +469,40 @@ def install_distribution(args, workspace, run_build_script):
 
     install[args.distribution](args, workspace, run_build_script)
 
+def install_boot_loader_arch(args, workspace):
+
+    mkinitcpio_old = os.path.join(workspace, "root", "etc/mkinitcpio.conf")
+    mkinitcpio_new = os.path.join(workspace, "root", "etc/mkinitcpio.conf.new")
+
+    with open(mkinitcpio_old, "r") as old:
+        with open(mkinitcpio_new, "w") as new:
+            for line in old:
+                if line.startswith("HOOKS="):
+                    new.write("HOOKS=\"systemd modconf block filesystems fsck\"\n")
+                else:
+                    new.write(line)
+
+    shutil.copystat(mkinitcpio_old, mkinitcpio_new)
+    os.remove(mkinitcpio_old)
+    shutil.move(mkinitcpio_new, mkinitcpio_old)
+
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "mkinitcpio", "-p", "linux"], check=True)
+
+    shutil.copyfile(os.path.join(workspace, "root", "boot/vmlinuz-linux"),
+                    os.path.join(workspace, "root", "boot/efi/vmlinuz-linux"))
+
+    shutil.copyfile(os.path.join(workspace, "root", "boot/initramfs-linux.img"),
+                    os.path.join(workspace, "root", "boot/efi/initramfs-linux.img"))
+
+    with open(os.path.join(workspace, "root", "boot/efi/loader/loader.conf"), "w") as f:
+        f.write("default arch\n")
+
+    with open(os.path.join(workspace, "root", "boot/efi/loader/entries/arch.conf"), "w") as f:
+        f.write("title   Arch Linux\n")
+        f.write("linux   /vmlinuz-linux\n")
+        f.write("initrd  /initramfs-linux.img\n")
+        f.write("options quiet rw\n")
```

Comment:
I really wished arch could just use kernel-install so that we don't have to write things manually like this...